### ISSUE TITLE
Add better atmosphere model for the Earth

### ISSUE THREAD

**Issue by astrojuanlu** [poliastro/poliastro#702]
(Coming from #694)

See for example:

* https://github.com/DeepHorizons/Python-NRLMSISE-00
* https://github.com/space-physics/msise00

(and probably others)

**Comment by Juanlu001** [comment]
This was (partially) addressed in #738. @jorgepiloto would you please leave a comment here stating what needs to be done, or alternatively close the issue?

**Comment by jorgepiloto** [comment]
Sure! I will make a brief resume on the current status of this:

After #738 was merged two new atmospheric models were added in `poliastro`:
* COESA62: up to 700km
* COESA76: up to 1000km

However, some difficulties appeared when implementing pressure and density in previous atmospheres above some altitudes. Different `#TODO` comments were left in the code to know what is still required to be done.

On the he other hand, more models could be added such as:
* Jacchia77
* NRLMSISE-00

This last model has been implemented in Python packages as stated at the beginning of this issue. However, I have tried to install [this package](https://github.com/space-physics/msise00) and it requires to have a `Fortran` compiler installed in user's machine.

Before closing this I would like to solve the following tasks:
- [ ] Full implemented COESA62
- [ ] Full implemented COESA76
- [ ] Full implemented Jacchia77 (could be included as dependence)
- [ ] Full implemented NRLMSISE-00 (could be included as dependence)

**Comment by jorgepiloto** [comment]
Just found this [amazing atmospheric models website](http://www.dem.inpe.br/~val/atmod/default.html) which holds source codes written in Fortran and C/C++ for different Jacchia's models and the famous MSIS-86. I will take a closer look in a few days :rocket:

**Comment by GorgiAstro** [comment]
If NRLMSISE-00 gets implemented, it would be nice to feed it with the daily space weather data combined by CSSI/AGI: ftp://ftp.agi.com/pub/DynamicEarthData/SpaceWeather-All-v1.2.txt and http://celestrak.com/SpaceData/SpaceWx-format.php

It contains three-hourly Ap geomagnetic index and daily F10.7 flux required by NRLMSISE-00.

The CSSI/AGI file is for instance also packaged into GMAT (in the folder `GMAT/R2019aBeta1/data/atmosphere/earth`). If we choose to package this file in the poliastro repo, I don't know what's the best way to keep this file up to date without needing to manually push new versions. This file is updated several times a day. Although if an extreme accuracy is not required, the file does not need to be updated very often as it offers predictions until 25 years in the future.

I am currently writing a reader for this file in Java (for Orekit), I could also write it in Python for poliastro: https://gitlab.orekit.org/orekit/orekit/-/issues/656

**Comment by priyanshurohilla** [comment]
I found a amazing python Package for [NRL MSISE-00 ](https://pypi.org/project/msise00/) atmospheric model, whicg we can include in poliastro as a dependency. They have covered altitude from 0-1000 km which is awesome!! Playing with it right now.. More insights coming up😁

**Comment by astrojuanlu** [comment]
Nice! I see that the project is quite big and includes a fair amount of FORTRAN code:

https://github.com/space-physics/msise00

**Comment by astrojuanlu** [comment]
Related: #833

**Comment by priyanshurohilla** [comment]
> Nice! I see that the project is quite big and includes a fair amount of FORTRAN code:
> 
> https://github.com/space-physics/msise00

So do u recommend this library for poliastro? Should we have it as a dependency?
If yes then i would start working on merging the codebase to poliastro. I have already tested this and i found it to be pretty accurate, the only problem which is clicking me is the requirement of fortran compiler which will add more complexity to poliastro testing!

**Comment by astrojuanlu** [comment]
> the only problem which is clicking me is the requirement of fortran compiler which will add more complexity to poliastro testing!

Yeah, I have the same feeling :cry: I think we should not carry such a heavy dependency which most users won't use. This is part of the [roadmap for Earth capabilities](https://github.com/poliastro/poliastro/projects/5), and we'll need to discuss the approach in #791 first.

In my opinion we might want to split some of this functionality into other package, but it's too early to tell.

**Comment by priyanshurohilla** [comment]
> > the only problem which is clicking me is the requirement of fortran compiler which will add more complexity to poliastro testing!
> 
> Yeah, I have the same feeling :cry: I think we should not carry such a heavy dependency which most users won't use. This is part of the [roadmap for Earth capabilities](https://github.com/poliastro/poliastro/projects/5), and we'll need to discuss the approach in #791 first.
> 
> In my opinion we might want to split some of this functionality into other package, but it's too early to tell.

@astrojuanlu hmm interesting.. but splitting the package may result in many new issues which we need to solve. Anyways we shall discuss it in the Roadmap😊

**Comment by astrojuanlu** [comment]
This was closed in #1113! 🎉

**Comment by DhruvSondhi** [comment]
@astrojuanlu we still need to implement NRLMSISE-00 ... I was thinking it would finish up all the implementation required 🤔

**Comment by astrojuanlu** [comment]
Oh, you're right, thanks @DhruvSondhi - Reopening!
